packages/x-date-pickers/src/DesktopTimePicker/DesktopTimePicker.tsx

```diff
@@ -56,9 +56,13 @@ const DesktopTimePicker = React.forwardRef(function DesktopTimePicker<TDate>(
   const actionBarActions: PickersActionBarAction[] = shouldRenderTimeInASingleColumn
     ? []
     : ['accept'];
-  const views: readonly TimeViewWithMeridiem[] = defaultizedProps.ampm
-    ? [...defaultizedProps.views, 'meridiem']
-    : defaultizedProps.views;
+  // Need to avoid adding the `meridiem` view when unexpected renderer is specified
+  const shouldHoursRendererContainMeridiemView =
+    viewRenderers.hours?.name === renderMultiSectionDigitalClockTimeView.name;
```

Comment:
If you have any better solutions—I'm welcome to suggestions 🙈 🤷

Reply:
Why can't you use the `shouldRenderTimeInASingleColumn`?

Reply:
Ok, that's triky. Maybe better to keep your hacky solution for now, And come back on this issue when documenting how to customize those view